src/main/java/org/owasp/validator/html/scan/AntiSamyDOMScanner.java

```diff
@@ -97,7 +97,28 @@ public AntiSamyDOMScanner(Policy policy) {
     public AntiSamyDOMScanner() throws PolicyException {
         super();
     }
-
+    // Method to decode the Unicode escape sequences
+    private String decodeUnicodeEscapes(String input) {  
```

Comment:
I guess import-related regex was introduced in https://github.com/Zimbra/antisamy/pull/4, and digging deeper found that it was introduced to circumvent an issue in antisamy library https://github.com/nahsra/antisamy/issues/24, and as per antisamy developer it's issue with CSS parser used in antisamy which seems fixed in https://github.com/nahsra/antisamy/issues/108
so if we upgrade the antisamy library then it's very well possible that we could remove our custom handling and also get rid of security issue

Reply:
I have further investigated this, and here are my findings:
The fix mentioned in https://github.com/nahsra/antisamy/issues/108 refers to enabling embedStyleSheets
`<directive name="embedStyleSheets" value="true"/>`

According to the documentation https://github.com/nahsra/antisamy/wiki/AntiSamy-Directives,
the embedStyleSheets directive allows external stylesheets referenced through @import to be fetched and embedded into the sanitized output. Allowing CSS imports from external URLs is a dangerous practice. It exposes the application to security risks by allowing the inclusion of potentially malicious external CSS, which goes against AntiSamy’s purpose of ensuring secure input sanitization. Support for this feature in AntiSamy is deprecated and will be removed in a future release.

As described in https://github.com/Zimbra/antisamy/pull/2, there remains an issue where media queries are stripped during sanitization. This behavior is attributed to the underlying third-party library (org.apache.xml.serialize.HTMLSerializer) used for document serialization within AntiSamy. The fix in https://github.com/nahsra/antisamy/issues/108 does not explicitly address this media query stripping issue.
Therefore, it is uncertain whether the upgrade resolves the media query serialization issue. 

Suggested Next Steps:
Conduct a detailed review of the latest AntiSamy release to verify if the media query stripping issue (linked to HTMLSerializer) has been resolved.
Determine whether the upgrade allows us to eliminate our custom handling without compromising security.
Ensure the embedStyleSheets directive remains disabled to mitigate risks associated with remote CSS imports.